include/seqan3/alphabet/concept.hpp

```diff
@@ -130,31 +128,19 @@ struct assign_rank_to_fn
     SEQAN3_CPO_IMPL(0, (v.assign_rank(args...)                               ))    // member
 
 public:
-    //!\brief Operator definition for lvalues.
-    template <typename alph_t>
-    //!\cond
-        requires requires (seqan3::alphabet_rank_t<alph_t> const r, alph_t & a)
-            { { impl(priority_tag<2>{}, a, r) }; }
-    //!\endcond
-    constexpr alph_t & operator()(seqan3::alphabet_rank_t<alph_t> const r, alph_t & a) const noexcept
-    {
-        static_assert(noexcept(impl(priority_tag<2>{}, a, r)),
-            "Only overloads that are marked noexcept are picked up by seqan3::assign_rank_to().");
-        static_assert(std::Same<alph_t &, decltype(impl(priority_tag<2>{}, a, r))>,
-            "The return type of your assign_rank_to() implementation must be 'alph_t &'.");
-
-        return impl(priority_tag<2>{}, a, r);
-    }
-
-    //!\brief Operator definition for rvalues.
+    //!\brief Operator definition.
     template <typename alph_t>
     //!\cond
         requires requires (seqan3::alphabet_rank_t<alph_t> const r, alph_t & a)
-            { { impl(priority_tag<2>{}, a, r) }; } && (!std::is_lvalue_reference_v<alph_t>)
+            {
```

Comment:
why was this before separated into lvalue and rvalues? Did something change semantically?

Reply:
I was too stupid to realise before that one overload would be enough :man_shrugging:

Reply:
ok